torch/csrc/jit/interpreter_autograd_function.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "torch/csrc/jit/interpreter.h"
+#include "torch/csrc/autograd/function.h"
+#include "torch/csrc/autograd/functions/utils.h"
+#include "torch/csrc/autograd/functions/basic_ops.h"
+namespace torch { namespace jit {
+struct InterpreterAutogradFunction : public autograd::Function {
+  InterpreterAutogradFunction(const jit::Function & function)
+  : interp_(function) {}
+  InterpreterAutogradFunction(const Interpreter & interp_, autograd::FunctionFlags && f)
+  : autograd::Function(std::move(f)), interp_(interp_) {}
+
+  virtual void willReleaseVariables() override {
+    keep_graph = false;
+  }
+  virtual autograd::variable_list apply(const autograd::variable_list& inputs) override {
+    std::vector<at::Tensor> tinputs;
+    std::vector<at::Tensor> toutputs;
+    for(auto & i : inputs) {
+      tinputs.push_back(i.data());
+    }
+    Interpreter interp = (keep_graph) ? interp_.clone() : interp_;
```

Comment:
This will call the copy constructor of `interp` - probably not what you wanted.

Reply:
interp is just a handle to a `std::shared_ptr<InterpreterImpl>` so copy constructor does not make a full copy. I'll think about whether there is a way to make this more clear.

Reply:
Ah right